code/modules/html_interface/map/station_map.dm

```diff
@@ -0,0 +1,82 @@
+
+/proc/generateHoloMinimaps()
+	var/list/filters = list(
+		"deathsquad",
+		"ert",
+		"nukeops",
+		"elitesyndicate",
+		"vox",
+		)
+
+	for (var/f in filters)
+		centcommminimaps |= f
+		generateCentcommMinimap(f)
+
+	for (var/z = 1 to world.maxz)
+		holominimaps |= z
+		generateHoloMinimap(z)
+
+/proc/generateHoloMinimap(var/zLevel=1)
+	var/icon/canvas = icon('icons/480x480.dmi', "blank")
+
+	if(zLevel != map.zCentcomm)//The Centcomm Zlevel's minimap will remain blank for immersion purposes. Until we add a way to only draw certain areas depending on the suit's faction
+		for(var/i = 1 to 480)
```

Comment:
I mean the latter, and you're right, so that's weird. I'll experiment a bit tonight.

Ah, It might be because the holomap's background was originally the screen object's icon. Before I made it an image as it is right now. Then there's no problems.